File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        long sum_ = 0;
        int l = 0, ans = nums.size() + 1;
        for (int i = 0; i < nums.size(); ++i){
            sum_ += nums[i];
            while(sum_ >= target){
                sum_ -= nums[l];
                ans = min(ans, i - l + 1);
                l += 1;
            }
        }
        if(ans == nums.size() + 1){
            return 0;
        }
        else{
            return (int)ans;
        }
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        sum_ = 0
        l = 0
        ans = len(nums) + 1
        for i, n in enumerate(nums):
            sum_ += n
            while sum_ >= target:
                sum_ -= nums[l]
                l += 1
                ans = min(ans, i - l + 2)
        if ans == len(nums) + 1:
            return 0
        return ans